File: backend/leave/dtos/response/leave_response_dto.py
class LeaveResponseDTO:
        
        @staticmethod
        def success(data,message='Success!'):
                return {
                        'success': True,
                        'message': message, 
                        'data': data,
                }
        
        @staticmethod
        def error(message='Somthing went wrong!'):
                return {
                        'success': False,
                        'message': message, 
                        'data': None,
                }
        
        @staticmethod
        def list_response(data):
                return {
                        'success': True,
                        'count': len(data), 
                        'data': data,
                }
                

File: frontend/src/dtos/response/leaveResponseDTO.js
export const parseLeaveListResponse=(data)=>({

    success: data.success,
    count: data.count,
    leaves: data.data,
});

export const parseSingleLeaveResponse=(data)=>({

    success: data.success,
    message: data.message,
    leave: data.data,
}); 